src/simulation/mod.rs

```diff
@@ -1,17 +1,34 @@
-use crate::config::Config;
-use crate::mpi::events::proto::Event;
-use crate::mpi::events::EventsPublisher;
-use crate::mpi::message_broker::{MessageBroker, MpiMessageBroker};
-use crate::mpi::messages::proto::leg::Route;
-use crate::mpi::messages::proto::{Activity, Agent, GenericRoute, Vehicle, VehicleType};
-use crate::mpi::population::Population;
-use crate::mpi::time_queue::TimeQueue;
-use crate::parallel_simulation::network::link::Link;
-use crate::parallel_simulation::network::network_partition::NetworkPartition;
-use crate::parallel_simulation::network::node::{ExitReason, NodeVehicle};
-use crate::parallel_simulation::routing::router::Router;
+use crate::simulation::config::Config;
+use crate::simulation::messaging::events::proto::Event;
+use crate::simulation::messaging::events::EventsPublisher;
+use crate::simulation::messaging::message_broker::{MessageBroker, MpiMessageBroker};
+use crate::simulation::messaging::messages::proto::leg::Route;
+use crate::simulation::messaging::messages::proto::{
+    Activity, Agent, GenericRoute, Vehicle, VehicleType,
+};
+use crate::simulation::network::link::Link;
+use crate::simulation::network::network_partition::NetworkPartition;
+use crate::simulation::network::node::{ExitReason, NodeVehicle};
+use crate::simulation::population::Population;
+use crate::simulation::routing::router::Router;
+use crate::simulation::time_queue::TimeQueue;
 use log::info;
 
+pub mod config;
+pub mod controller;
+mod id_mapping;
+pub mod io;
+pub mod logging;
+pub mod messaging;
+#[allow(dead_code)]
+mod network;
+#[allow(dead_code)]
+mod partition_info;
+mod population;
+#[allow(unused_comparisons, dead_code)]
+mod routing;
+pub mod time_queue;
+
 pub struct Simulation<'sim> {
```

Comment:
I think it would make sense to have this `mod.rs` file here with the above module definitions as well as a `simulation.rs` file with the simulation below. This way we would have one place with module and visibility stuff and one place which runs a simulation

Reply:
Done. Also preserved git history (https://stackoverflow.com/questions/16937359/git-copy-file-preserving-history)

Reply:
Nice